### PR TITLE
Added validation for support types

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/31309

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 1]
```diff
@@ -208,3 +208,24 @@ def test_is_approved_tags(self, repo, tags, is_valid):
...
             assert 'The pack metadata contains non approved tags: NonApprovedTag' in self.validator.get_errors()
+
+    @pytest.mark.parametrize('type, is_valid', [
+        ('community', True),
+        ('partner', True),
+        ('xsoar', True),
+        ('someName', False),
+        ('test', False)
+    ])
+    def test_is_valid_support_type(self, repo, type, is_valid):
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:219]: please add docstring

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 SUPPORTED_CONTRIBUTORS_LIST = ['partner', 'developer']
 ISO_TIMESTAMP_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
 ALLOWED_CERTIFICATION_VALUES = ['certified', 'verified']
+SUPPORT_TYPES = ['partner', 'community', 'xsoar']
```
Comment (Itay4) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:34]:
```suggestion
SUPPORT_TYPES = ['community', 'xsoar'] + SUPPORTED_CONTRIBUTORS_LIST
```

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 3]
```diff
@@ -208,3 +208,24 @@ def test_is_approved_tags(self, repo, tags, is_valid):
         assert self.validator._is_approved_tags() == is_valid
         if not is_valid:
             assert 'The pack metadata contains non approved tags: NonApprovedTag' in self.validator.get_errors()
+
+    @pytest.mark.parametrize('type, is_valid', [
+        ('community', True),
+        ('partner', True),
+        ('xsoar', True),
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:215]: please also add test for `developer`

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py (hunk 2 above)
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:34]: I think it is better to use `PACK_SUPPORT_OPTIONS` in constants.py (and rm developer)

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -949,6 +950,11 @@ def pack_metadata_isnt_json(pack_meta_file):
     def pack_metadata_missing_url_and_email():
         return 'Contributed packs must include email or url.'
 
+    @staticmethod
+    @error_code_decorator
+    def pack_metadata_invalid_support_type(pack_meta_file):
+        return 'Support field should be one of the following: `xsoar`, `partner` or `community`.'
```
Comment (Itay4) [inline demisto_sdk/commands/common/errors.py:956]: what are the backticks for? what does it look like in the cli?

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 * Fixed an issue with support for Docker Desktop on Mac version 2.5.0+.
 * Added support for vulture and mypy linting when running without docker.
 * Improved retry support when building docker images for linting.
+* Added validation for support types in pack_metadata file.
```
Comment (Itay4) [inline CHANGELOG.md:6]:
```suggestion
* Added validation for support types in the pack metadata file.
```

File: demisto_sdk/commands/common/errors.py (hunk 4 above)
Comment (Shellyber) [reply demisto_sdk/commands/common/errors.py:956]: it looks like this:
image

Remove?

Comment (Itay4) [reply demisto_sdk/commands/common/errors.py:956]: yea

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py (hunk 3 above)
Comment (bakatzir) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py:215]: @ShahafBenYakir isn't the pack support of type developer deprecated?

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 6]
```diff
@@ -208,3 +208,38 @@ def test_is_approved_tags(self, repo, tags, is_valid):
...
+    def test_is_valid_support_type(self, repo, type, is_valid):
+        """
+        Given:
+            - Pack with support type in the metadata file.
+
+        When:
+            - Running _is_valid_support_type.
+
+        Then:
+            - Case A: Ensure validation passes as there are no usecases to verify
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:229]: please update docstring with the relevant cases..

**Comment by Shellyber** [comment]
Can only be merged after https://github.com/demisto/content/pull/10270

File: demisto_sdk/commands/common/errors.py [hunk 7]
```diff
@@ -949,6 +950,11 @@ def pack_metadata_isnt_json(pack_meta_file):
     def pack_metadata_missing_url_and_email():
         return 'Contributed packs must include email or url.'
 
+    @staticmethod
+    @error_code_decorator
+    def pack_metadata_invalid_support_type(pack_meta_file):
+        return 'Support field should be one of the following: xsoar, partner or community.'
```
Comment (Itay4) [inline demisto_sdk/commands/common/errors.py:956]:
```suggestion
        return 'Support field should be one of the following: xsoar, partner, developer or community.'
```

**Review by Itay4 — APPROVED** [review]

**Pull request merged**